test/SignAndDelete.test.js

```diff
@@ -0,0 +1,308 @@
+jest.mock("../src/HttpResponses");
+jest.mock("aws-sdk", () => {
+  const KMS = class {};
+  KMS.prototype.decrypt = jest.fn();
+  return { KMS };
+});
+
+describe("SignAndDelete", () => {
+  let chance;
+  let underTest;
+
+  const setUp = () => {
+    chance = Chance();
+
+    const { decrypt } = KMS.prototype;
+    decrypt.mockImplementation(({ CiphertextBlob }) => {
+      return {
+        promise() {
+          return Promise.resolve({ Plaintext: CiphertextBlob });
+        },
+      };
+    });
+
+    jest.spyOn(OAuth, "OAuth").mockImplementation(() => ({
+      delete: (link, accessToken, accessTokenSecret, callback) => {
+        callback(null, null, { statusCode: 200 });
+      },
+    }));
+
+    getConfig.mockImplementation(() => {
+      return {};
+    });
+
+    underTest = require("../src/SignAndDelete");
+  };
+
+  beforeEach(async () => {
+    setUp();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("config has values defined", () => {
+    let expectedFirstLegUri;
+    let expectedThirdLegUri;
+    let expectedClientKey;
+    let expectedClientSecret;
+    let expectedOAuthVersion;
+    let expectedAuthorizeCallbackUri;
+    let expectedOAuthSignatureMethod;
+    let expectedOAuthNonceSize;
+
+    beforeEach(async () => {
+      expectedFirstLegUri = chance.string();
+      expectedThirdLegUri = chance.string();
+      expectedClientKey = chance.string();
+      expectedClientSecret = chance.string();
+      expectedOAuthVersion = chance.string();
+      expectedAuthorizeCallbackUri = chance.string();
+      expectedOAuthSignatureMethod = chance.string();
+      expectedOAuthNonceSize = chance.string();
+
+      getConfig.mockImplementation(() => {
+        return {
+          firstLegUri: expectedFirstLegUri,
+          thirdLegUri: expectedThirdLegUri,
+          clientKey: expectedClientKey,
+          clientSecret: expectedClientSecret,
+          oAuthVersion: expectedOAuthVersion,
+          authorizeCallbackUri: expectedAuthorizeCallbackUri,
+          oAuthSignatureMethod: expectedOAuthSignatureMethod,
+          oAuthNonceSize: expectedOAuthNonceSize,
+        };
+      });
+
+      await underTest.doSignAndDelete(
+        chance.string(),
+        chance.string(),
+        chance.string()
+      );
+    });
+
+    it("creates an OAuth with same values from config", () => {
+      expect(OAuth.OAuth).toBeCalledWith(
+        expectedFirstLegUri,
+        expectedThirdLegUri,
+        expectedClientKey,
+        expectedClientSecret,
+        expectedOAuthVersion,
+        expectedAuthorizeCallbackUri,
+        expectedOAuthSignatureMethod,
+        expectedOAuthNonceSize,
+        expect.anything()
+      );
+    });
+  });
+
+  describe("is called", () => {
+    beforeEach(async () => {
+      await underTest.doSignAndDelete(
+        chance.string(),
+        chance.string(),
+        chance.string()
+      );
+    });
+
+    it("passed JD Accept header", () => {
```

Comment:
```suggestion
    it("should have the JD Accept header", () => {
```

Or something like that